testerum-backend/testerum-runner/testerum-runner-junit/src/main/java/com/testerum/runner/junit5/TesterumJunitTestFactory.java

```diff
@@ -0,0 +1,87 @@
+package com.testerum.runner.junit5;
+
+import static com.testerum.runner_cmdline.cmdline.params.parser.TestPathParserKt.parseStringToTestPath;
+
+import com.testerum.model.tests_finder.TestPath;
+import com.testerum.runner.junit5.provider.JunitTestProvider;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.DynamicNode;
+
+public class TesterumJunitTestFactory {
+
+    private final Path repositoryDirectory;
+    private String variablesEnvironment;
+    private Map<String, String> variableOverrides = Collections.emptyMap();
+    private Path settingsFile;
+    private Map<String, String> settingOverrides = Collections.emptyMap();
+    private List<TestPath> testPaths = Collections.emptyList();
+    private List<String> tagsToInclude = Collections.emptyList();
+    private List<String> tagsToExclude = Collections.emptyList();
+
+    public TesterumJunitTestFactory(String repositoryDirectory) {
```

Comment:
add ``NotNull`` and ``Nullable`` (from the package ``org.jetbrains.annotations``) annotations:
* for all method return types
* for all method parameters

Advantages:
* API documentation for people using our code
* IntelliJ warnings in the IDE when doing wrong stuff
* simplified usage from Kotlin (the Kotlin compiler knows what to expect)